atom/browser/api/atom_api_net_log.cc

```diff
@@ -40,22 +48,64 @@ void NetLog::StartLogging(mate::Arguments* args) {
     return;
   }
 
-  net_log_->StartDynamicLogging(log_path);
+  // TODO(deepak1556): Provide more flexibility to this module
+  // by allowing customizations on the capturing options.
+  net_log_writer_->StartNetLog(
+      log_path, net::NetLogCaptureMode::Default(),
+      net_log::NetExportFileWriter::kNoLimit /* file size limit */,
+      base::CommandLine::ForCurrentProcess()->GetCommandLineString(),
+      std::string(), {} /* URLRequestContextGetter list */);
+}
+
+std::string NetLog::GetLoggingState() const {
+  const base::Value* current_log_state =
+      net_log_state_->FindKeyOfType("state", base::Value::Type::STRING);
+  if (!current_log_state)
+    return std::string();
+  return current_log_state->GetString();
 }
 
-bool NetLog::IsCurrentlyLogging() {
-  return net_log_->IsDynamicLogging();
+bool NetLog::IsCurrentlyLogging() const {
+  return (GetLoggingState() == "LOGGING");
 }
 
-base::FilePath::StringType NetLog::GetCurrentlyLoggingPath() {
-  return net_log_->GetDynamicLoggingPath().value();
+std::string NetLog::GetCurrentlyLoggingPath() const {
+  // Net log exporter has a default path which will be used
+  // when no log path is provided, but since we don't allow
+  // net log capture without user provided file path, this
+  // check is completely safe.
+  if (IsCurrentlyLogging()) {
+    const base::Value* current_log_path =
+        net_log_state_->FindKeyOfType("file", base::Value::Type::STRING);
+    if (current_log_path)
+      return current_log_path->GetString();
+  }
+
+  return std::string();
 }
 
 void NetLog::StopLogging(mate::Arguments* args) {
   base::OnceClosure callback;
   args->GetNext(&callback);
+  if (IsCurrentlyLogging()) {
+    stop_callback_queue_.emplace_back(std::move(callback));
+    net_log_writer_->StopNetLog(nullptr, nullptr);
+  } else {
+    std::move(callback).Run();
+  }
+}
+
+void NetLog::OnNewState(const base::DictionaryValue& state) {
+  net_log_state_ = state.CreateDeepCopy();
 
-  net_log_->StopDynamicLogging(std::move(callback));
+  if (stop_callback_queue_.empty())
```

Comment:
This doesn't matter right now, because we're not doing much here; but ☝️  should be checked inside the `if` block 👇 in the stop logging case or it can be removed.